src/psbt/finalizer.rs

```diff
@@ -268,26 +270,17 @@ pub fn finalize_helper<C: secp256k1::Verification>(
 
     // Check well-formedness of input data
     for (n, input) in psbt.inputs.iter().enumerate() {
-        let target = input.sighash_type.unwrap_or(bitcoin::SigHashType::All);
-        for (key, rawsig) in &input.partial_sigs {
-            if rawsig.is_empty() {
-                return Err(Error::InputError(
-                    InputError::InvalidSignature {
-                        pubkey: *key,
-                        sig: rawsig.clone(),
-                    },
-                    n,
-                ));
-            }
-            let (flag, sig) = rawsig.split_last().unwrap();
-            let flag = bitcoin::SigHashType::from_u32_standard(*flag as u32).map_err(|_| {
-                super::Error::InputError(
-                    InputError::Interpreter(interpreter::Error::NonStandardSigHash(
-                        [sig, &[*flag]].concat().to_vec(),
-                    )),
-                    n,
-                )
-            })?;
+        let target = input.sighash_type.unwrap_or(bitcoin::EcdsaSigHashType::All);
+        for (key, ecdsa_sig) in &input.partial_sigs {
+            let flag = bitcoin::EcdsaSigHashType::from_u32_standard(ecdsa_sig.hash_ty as u32)
```

Comment:
In f3c38b8cc04fed0a68f4d6074d8c30f6912d958f:

lol, is this conversion just to check standardness? We should add a method to rust-bitcoin so we can call `ecdsa_sig.hash_tx.is_standard()'.

Reply:
Actually, this line is not needed. ecdsa_sig.hash_ty is guaranteed to follow standardness rules.